homeassistant/components/alarm_control_panel/__init__.py

```diff
@@ -109,6 +110,11 @@ def code_format(self):
         """Regex for code format or None if no code is required."""
         return None
 
+    @property
+    def code_format_per_user(self):
```

Comment:
This makes changes to our entity-component models and requires an architectural discussion before creating this change.

See: <https://developers.home-assistant.io/docs/core/entity/#changing-the-entity-model>

Reply:
Thanks for the quick response, I have opened a discussion thread, [see here](https://github.com/home-assistant/architecture/issues/460)
Although in my past I've been a software engineer, I've been out of main line development for several years. Also, my version control experience is with Mercurial and not Git so please forgive me when Imake mistakes.